packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -81,7 +106,16 @@ class IMobileDevice {
 
   Future<String> getInfoForDevice(String deviceID, String key) async {
     try {
-      final ProcessResult result = await processManager.run(<String>['ideviceinfo', '-u', deviceID, '-k', key]);
+      final ProcessResult result = await processManager.run(
+        <String>[
+          cache.getArtifactFile('libimobiledevice', 'ideviceinfo'),
```

Comment:
Leave a comment here for tracking. As we talked earlier, it would be easier to replace the tools internally if we:
1. Use an `enum` instead of a `String` to find the tools
2. Temporarily fallback to use the tools in `$PATH` if file is not found, to avoid breaking existing users internally during the transition. After the transition we could remove the fallback.

Reply:
done